crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3782,10 +3776,10 @@ impl<'db> TypeInferenceBuilder<'db> {
 
 /// Type expressions
 impl<'db> TypeInferenceBuilder<'db> {
-    fn infer_type_expression(&mut self, expression: &ast::Expr) -> Type<'db> {
+    fn infer_type_expression_impl(&mut self, expression: &ast::Expr) -> Type<'db> {
```

Comment:
I had to split up `infer_type_expression` into two methods, `infer_type_expression` and `infer_type_expression_impl`, or `cargo fmt` insisted on increasing the indentation of the huge `match` statement in `infer_type_expression` with my changes here. I didn't want a huge diff in this PR, and I also didn't think the extra indentation was desirable in and of itself, so I split the method into two.

Reply:
Oh, and now I see @sharkdp does a very similar thing in https://github.com/astral-sh/ruff/pull/14106/files#r1829286623. Let's merge his PR first; I can rebase this on top.